2-projects/1-todo-list/Extra Advanced Challenge - DUE DATE/script.js

```diff
@@ -0,0 +1,223 @@
+                  <i class="fa fa-check" aria-hidden="true"></i>
+                  <i class="fa fa-trash" aria-hidden="true"></i>
+                </span>
+*/
+
+        let myNewSpan = document.createElement("span");
+        // Apply the necessary classes
+        myNewSpan.classList.add(...spanClassesToAdd);       
+
+        let myNewIcon = document.createElement("i");
+        // Apply the necessary classes
+        myNewIcon.classList.add("fa","fa-check");
+        myNewIcon.ariaHidden=true;
+        // Each 'fa-check <i>' should have an eventListener that will listen for clicks. 
+        myNewIcon.addEventListener("click", checkFunction);
+        // Add this to the '<span>'.
+        myNewSpan.appendChild(myNewIcon);
+
+        myNewIcon = document.createElement("i");
+        // Apply the necessary classes
+        myNewIcon.classList.add("fa","fa-trash");
+        myNewIcon.ariaHidden=true;
+        // In my opinion, these icons are too close together - so I added some padding
+        myNewIcon.style.paddingLeft="1em";   
+
+        // Each 'fa-trash <i>' should have an eventListener that will listen for clicks. 
+        myNewIcon.addEventListener("click", trashFunction);
+        // Add this to the '<span>'.
+        myNewSpan.appendChild(myNewIcon);
+
+        // Add this to the '<li>'.
+        listViewItem.appendChild(myNewSpan);
+
+        // Append to the list
+        list.appendChild(listViewItem);
+  }
+
+}
+
+
+// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. 
+// These new todos will need the completed and delete buttons adding like normal.
+
+
+/*
+the parsed value of a input date is always formatted yyyy-mm-dd.
+dateControl.value = '2017-06-01';
+
+console.log(dateControl.value); // prints "2017-06-01"
+console.log(dateControl.valueAsNumber); // prints 1496275200000, a JavaScript timestamp (ms)
+*/
+
+
+function addNewTodo(event) {
+  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
+  event.preventDefault();
+  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let newToDoInput = document.getElementById("todoInput")
+  let inputText = newToDoInput.value.trim();
+  newToDoInput.value = "" // reset the input field to blank 
+  let theDateInfo = document.getElementById("taskdate");
+  if (inputText) // definitely non-null
+  {
+    // Check whether a date has been entered
+    // If so attach it to the 'todo' otherwise ignore
+    let newTodoObj = {task: inputText, completed: false};
+    if (theDateInfo.valueAsNumber >= tomorrowAsNumber) 
+    {
+    //  a date has been entered
+        newTodoObj.duedate = theDateInfo.valueAsNumber;
+        newTodoObj.task += " - Due by " + new Date(theDateInfo.value).toDateString();
+    }  
+        
+    populateTodoList([newTodoObj]) // Add new Todo Object
+  }
+  
+  else
```

Comment:
it's quite difficult to read this sort of if/else statement as it's not properly indented. This is how we would do it based on the guideline:
```
if () {
    ...
} else if () {
    ...
} else {
    ...
}
```